plugins/dbviewer/frontend/public/javascripts/countly.views.js

```diff
@@ -301,6 +314,33 @@ window.DBViewerView = countlyView.extend({
         });
 
         $('#alert-name-input').attr('placeholder', $.i18n.map["alert.enter-alert-name"]);
+        if (countlyGlobal.member.global_admin) {
+            $('#show-mongotop-mongostat-buttons').show();
+        }
+        $("#show-mongostat").on("click", function() {
+            self.getMongoStatResults();
+        });
+        $("#show-mongotop").on("click", function() {
+            self.getMongoTopResults();
+        });
+    },
+    getMongoTopResults: function() {
+        var self = this;
+        countlyDBviewer.getMongoTopData(function(res) {
+            if (res) {
+                app.navigate('#/manage/mongotop', false);
```

Comment:
move the navigation to the event handler above.
```
$("#show-mongotop").on("click", function() {
            app.navigate('#/manage/mongotop', false);
            self.getMongoTopResults();
});
```

because you call `getMongoTopResults` from the refresh function as well so you wouldn't want to renavigate.